docs/Reference-for-template.json.md

```diff
@@ -52,31 +52,275 @@ You can find a sample of a complex `template.json` file here [Sample](https://gi
+|`defaultValue`|The value assigned to the symbol if no value for it is provided by the user or host.|
+|`binding`|The name of the host property to take the value from.|	
+|`replaces`|The text to replace with the value of this symbol.|	 
+|`fileRename`|The fileRename element defines the filenames which will be replaced by symbol value.| 
+|`description`|Human readable text describing the meaning of the symbol. This has no effect on template generation.|
+|`isRequired`|Indicates if the parameter is required or not.|
+|`choices`|List of available choices. Applicable only when `datatype=choice.` Contains array of the elements: <br />- `choice`: possible value of the symbol.<br />- `description`: human readable text describing the meaning of the choice. This has no effect on template generation. <br /> If not provided, there are no valid choices for the symbol, so it can never be assigned a value.|
+|`onlyIf`| |
+|`forms`|Defines the set of transforms that can be referenced by symbol definitions. Forms allow the specification of a "replaces"/"replacement" pair to also apply to other ways the "replaces" value may have been specified in the source by specifying a transform from the original value of "replaces" in configuration to the one that may be found in the source. [Details](https://github.com/dotnet/templating/wiki/Runnable-Project-Templates---Value-Forms)|
+
+##### Examples
+Boolean optional parameter with default value `false`:
+```json
+  "symbols": {
+    "TestProject": {
+      "type": "parameter",
+      "dataType": "bool",
+      "defaultValue": "false"
+    }
+  },
+```
+ 
+Choice optional parameter with 3 possible values:
+```json
+  "symbols": {
+    "Framework": {
+      "type": "parameter",
+      "description": "The target framework for the project.",
+      "datatype": "choice",
+      "choices": [
+        {
+          "choice": "netcoreapp3.1",
+          "description": "Target netcoreapp3.1"
+        },
+        {
+          "choice": "netstandard2.1",
+          "description": "Target netstandard2.1"
+        },
+        {
+          "choice": "netstandard2.0",
+          "description": "Target netstandard2.0"
+        }
+      ],
+      "replaces": "netstandard2.0",
+      "defaultValue": "netstandard2.0"
+    }
+}
+```
+ 
+String optional parameter, replaces TargetFrameworkOverride:
+```json
+ "symbols": {
+    "TargetFrameworkOverride": {
+      "type": "parameter",
+      "description": "Overrides the target framework",
+      "replaces": "TargetFrameworkOverride",
+      "datatype": "string",
+      "defaultValue": ""
+    }
+}
+```
+
+#### Derived symbol
```

Comment:
By code, derived symbol can also have the following fields: `binding`, `defaultValue`, `isRequired`, `dataType`, `onlyIf`, `forms`. Does they make any sense for derived symbols?

Reply:
How do we figure out if removing any of these would be a breaking change. @sayedihashimi may be able to check NuGet, but how much will that cover? Are all of the inbox templates now on NuGet.

Reply:
I didn't see they were ever used in any example, they are not part of template.json schema. However, they are available from code. 
I don't suggest to do any change to remove them from code, just thinking that probably we should not document them unless there is any reason why they might be used with `derived` symbol.